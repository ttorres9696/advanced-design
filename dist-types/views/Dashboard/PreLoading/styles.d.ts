declare const _default: (props?: any) => Record<"img" | "root" | "loadingImg" | "cancelImg" | "loadingSubtitle" | "cancelButton", string>;
export default _default;
