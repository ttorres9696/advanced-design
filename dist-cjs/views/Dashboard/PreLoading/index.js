"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _core = require("@material-ui/core");

var _clsx2 = _interopRequireDefault(require("clsx"));

var _react = _interopRequireWildcard(require("react"));

var _reactRedux = require("react-redux");

var _actions = require("../../../redux/core/actions");

var _styles = _interopRequireDefault(require("./styles"));

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { if (typeof Symbol === "undefined" || !(Symbol.iterator in Object(arr))) return; var _arr = []; var _n = true; var _d = false; var _e = undefined; try { for (var _i = arr[Symbol.iterator](), _s; !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

var PreLoading = function PreLoading() {
  var _clsx;

  var classes = (0, _styles.default)();
  var dispatch = (0, _reactRedux.useDispatch)();
  var cancelledPreLoading = (0, _reactRedux.useSelector)(function (state) {
    return state.core.canceledPreLoading;
  });

  var _useState = (0, _react.useState)("LOADING INSTANT DESIGN..."),
      _useState2 = _slicedToArray(_useState, 2),
      message = _useState2[0],
      setMessage = _useState2[1];

  var handleClickCancelButton = (0, _react.useCallback)(function () {
    setMessage('INSTANT DESIGN CANCELLED');
    dispatch((0, _actions.cancelPreLoading)());
  }, [dispatch, setMessage]);
  return /*#__PURE__*/_react.default.createElement(_core.Box, {
    className: classes.root
  }, /*#__PURE__*/_react.default.createElement("i", {
    className: (0, _clsx2.default)(classes.img, (_clsx = {}, _defineProperty(_clsx, classes.loadingImg, !cancelledPreLoading), _defineProperty(_clsx, classes.cancelImg, cancelledPreLoading), _clsx))
  }), /*#__PURE__*/_react.default.createElement(_core.Typography, {
    variant: "overline",
    gutterBottom: true,
    className: classes.loadingSubtitle
  }, message), !cancelledPreLoading ? /*#__PURE__*/_react.default.createElement(_core.Typography, {
    variant: "overline",
    className: classes.cancelButton,
    onClick: handleClickCancelButton
  }, "CANCEL") : null);
};

var _default = /*#__PURE__*/_react.default.memo(PreLoading);

exports.default = _default;