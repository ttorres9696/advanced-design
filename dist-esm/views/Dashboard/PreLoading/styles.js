"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _styles = require("@material-ui/core/styles");

var _instantDesignCancelled = require("../../../shared/assets/instantDesignCancelled");

var _instantProposalLoading = require("../../../shared/assets/instantProposalLoading");

var _default = (0, _styles.makeStyles)(function (theme) {
  return (0, _styles.createStyles)({
    root: {
      height: '100vh',
      width: '100vw',
      backgroundColor: '#FAFAFA',
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'center',
      alignItems: 'center'
    },
    img: {
      width: 206,
      height: 133,
      display: 'inline-block',
      backgroundSize: 'contain',
      backgroundRepeat: 'no-repeat'
    },
    loadingImg: {
      backgroundImage: "url('".concat(_instantProposalLoading.instantProposalLoading, "')")
    },
    cancelImg: {
      backgroundImage: "url('".concat(_instantDesignCancelled.instantDesignCancelled, "')")
    },
    loadingSubtitle: {
      fontWeight: 'bold',
      fontSize: 14
    },
    cancelButton: {
      fontWeight: 'bold',
      fontSize: 12,
      cursor: 'pointer',
      color: '#066BB2'
    }
  });
});

exports.default = _default;