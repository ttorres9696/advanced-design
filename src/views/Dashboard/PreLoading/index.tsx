import { Box, Typography } from '@material-ui/core';
import clsx from 'clsx';
import React, { useCallback, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import { RootState } from '../../../redux';
import { cancelPreLoading } from '../../../redux/core/actions';
import useStyles from './styles';

const PreLoading: React.FC = () => {
  const classes = useStyles();

  const dispatch = useDispatch();
  const cancelledPreLoading = useSelector((state: RootState) => state.core.canceledPreLoading);

  const [message, setMessage] = useState<string>(`LOADING INSTANT DESIGN...`);

  const handleClickCancelButton = useCallback(() => {
    setMessage('INSTANT DESIGN CANCELLED');
    dispatch(cancelPreLoading());
  }, [dispatch, setMessage]);

  return (
    <Box className={classes.root}>
      <i
        className={clsx(classes.img, {
          [classes.loadingImg]: !cancelledPreLoading,
          [classes.cancelImg]: cancelledPreLoading,
        })}
      />

      <Typography variant="overline" gutterBottom className={classes.loadingSubtitle}>
        {message}
      </Typography>

      {!cancelledPreLoading ? (
        <Typography variant="overline" className={classes.cancelButton} onClick={handleClickCancelButton}>
          CANCEL
        </Typography>
      ) : null}
    </Box>
  );
};

export default React.memo(PreLoading);
