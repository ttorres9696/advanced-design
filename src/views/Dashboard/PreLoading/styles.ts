import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';

import { instantDesignCancelled } from '../../../shared/assets/instantDesignCancelled';
import { instantProposalLoading } from '../../../shared/assets/instantProposalLoading';

export default makeStyles((theme: Theme) =>
  createStyles({
    root: {
      height: '100vh',
      width: '100vw',
      backgroundColor: '#FAFAFA',
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'center',
      alignItems: 'center',
    },
    img: {
      width: 206,
      height: 133,
      display: 'inline-block',
      backgroundSize: 'contain',
      backgroundRepeat: 'no-repeat',
    },
    loadingImg: {
      backgroundImage: `url('${instantProposalLoading}')`,
    },
    cancelImg: {
      backgroundImage: `url('${instantDesignCancelled}')`,
    },
    loadingSubtitle: {
      fontWeight: 'bold',
      fontSize: 14,
    },
    cancelButton: {
      fontWeight: 'bold',
      fontSize: 12,
      cursor: 'pointer',
      color: '#066BB2',
    },
  }),
);
