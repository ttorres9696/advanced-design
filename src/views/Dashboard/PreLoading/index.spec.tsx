import { mount, ReactWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { Provider } from 'react-redux';

import mockStore from '../../../redux/specs/mocks/state.mock';
import PreLoading from '.';

let wrapper: ReactWrapper;

beforeEach(() => {
  wrapper = mount(
    <Provider store={mockStore}>
      <PreLoading />
    </Provider>,
    {},
  );
});

describe('PreLoading', () => {
  it('renders correctly', () => expect(toJson(wrapper)).toMatchSnapshot());
});
