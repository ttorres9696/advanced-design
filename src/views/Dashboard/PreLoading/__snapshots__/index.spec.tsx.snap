// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreLoading renders correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Memo(PreLoading)>
    <Styled(MuiBox)
      className="makeStyles-root-1"
    >
      <div
        className="MuiBox-root MuiBox-root-6 makeStyles-root-1"
      >
        <img
          alt=""
          className="makeStyles-img-2"
          src="instant_proposal_loading.gif"
        />
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-loadingSubtitle-3"
          gutterBottom={true}
          variant="overline"
        >
          <ForwardRef(Typography)
            className="makeStyles-loadingSubtitle-3"
            classes={
              Object {
                "alignCenter": "MuiTypography-alignCenter",
                "alignJustify": "MuiTypography-alignJustify",
                "alignLeft": "MuiTypography-alignLeft",
                "alignRight": "MuiTypography-alignRight",
                "body1": "MuiTypography-body1",
                "body2": "MuiTypography-body2",
                "button": "MuiTypography-button",
                "caption": "MuiTypography-caption",
                "colorError": "MuiTypography-colorError",
                "colorInherit": "MuiTypography-colorInherit",
                "colorPrimary": "MuiTypography-colorPrimary",
                "colorSecondary": "MuiTypography-colorSecondary",
                "colorTextPrimary": "MuiTypography-colorTextPrimary",
                "colorTextSecondary": "MuiTypography-colorTextSecondary",
                "displayBlock": "MuiTypography-displayBlock",
                "displayInline": "MuiTypography-displayInline",
                "gutterBottom": "MuiTypography-gutterBottom",
                "h1": "MuiTypography-h1",
                "h2": "MuiTypography-h2",
                "h3": "MuiTypography-h3",
                "h4": "MuiTypography-h4",
                "h5": "MuiTypography-h5",
                "h6": "MuiTypography-h6",
                "noWrap": "MuiTypography-noWrap",
                "overline": "MuiTypography-overline",
                "paragraph": "MuiTypography-paragraph",
                "root": "MuiTypography-root",
                "srOnly": "MuiTypography-srOnly",
                "subtitle1": "MuiTypography-subtitle1",
                "subtitle2": "MuiTypography-subtitle2",
              }
            }
            gutterBottom={true}
            variant="overline"
          >
            <span
              className="MuiTypography-root makeStyles-loadingSubtitle-3 MuiTypography-overline MuiTypography-gutterBottom"
            >
              LOADING INSTANT DESIGN...
            </span>
          </ForwardRef(Typography)>
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-cancelButton-4"
          onClick={[Function]}
          variant="overline"
        >
          <ForwardRef(Typography)
            className="makeStyles-cancelButton-4"
            classes={
              Object {
                "alignCenter": "MuiTypography-alignCenter",
                "alignJustify": "MuiTypography-alignJustify",
                "alignLeft": "MuiTypography-alignLeft",
                "alignRight": "MuiTypography-alignRight",
                "body1": "MuiTypography-body1",
                "body2": "MuiTypography-body2",
                "button": "MuiTypography-button",
                "caption": "MuiTypography-caption",
                "colorError": "MuiTypography-colorError",
                "colorInherit": "MuiTypography-colorInherit",
                "colorPrimary": "MuiTypography-colorPrimary",
                "colorSecondary": "MuiTypography-colorSecondary",
                "colorTextPrimary": "MuiTypography-colorTextPrimary",
                "colorTextSecondary": "MuiTypography-colorTextSecondary",
                "displayBlock": "MuiTypography-displayBlock",
                "displayInline": "MuiTypography-displayInline",
                "gutterBottom": "MuiTypography-gutterBottom",
                "h1": "MuiTypography-h1",
                "h2": "MuiTypography-h2",
                "h3": "MuiTypography-h3",
                "h4": "MuiTypography-h4",
                "h5": "MuiTypography-h5",
                "h6": "MuiTypography-h6",
                "noWrap": "MuiTypography-noWrap",
                "overline": "MuiTypography-overline",
                "paragraph": "MuiTypography-paragraph",
                "root": "MuiTypography-root",
                "srOnly": "MuiTypography-srOnly",
                "subtitle1": "MuiTypography-subtitle1",
                "subtitle2": "MuiTypography-subtitle2",
              }
            }
            onClick={[Function]}
            variant="overline"
          >
            <span
              className="MuiTypography-root makeStyles-cancelButton-4 MuiTypography-overline"
              onClick={[Function]}
            >
              CANCEL
            </span>
          </ForwardRef(Typography)>
        </WithStyles(ForwardRef(Typography))>
      </div>
    </Styled(MuiBox)>
  </Memo(PreLoading)>
</Provider>
`;
